yt/data_objects/data_containers.py

```diff
@@ -793,9 +795,18 @@ def create_firefly_object(
         ## for safety, in case someone passes a float just cast it
         default_decimation_factor = int(default_decimation_factor)
 
+        if JSONdir is not None:
+            issue_deprecation_warning(
+                "The 'JSONdir' keyword argument is a deprecated alias for 'datadir'."
+                "Please use 'datadir' directly.",
+                since="4.1",
+                removal="4.2",
```

Comment:
no need to specify a removal version, we'll try to keep this for a couple minor versions at least, and possibly longer if it's not a burden to maintain
```suggestion
```